pkgs/development/interpreters/nickel/default.nix

```diff
@@ -8,26 +8,33 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "nickel";
-  version = "1.1.1";
+  version = "1.2.1";
 
   src = fetchFromGitHub {
     owner = "tweag";
     repo = pname;
     rev = "refs/tags/${version}";
-    hash = "sha256-bG0vNfKQpFQHDBfokvTpfXgVmKg6u/BcIz139pLwwsE=";
+    hash = "sha256-iHHZ2CXle8edJoJDIOMrUNucTdhyNZpSKfAPUmnt6eI=";
   };
 
-  cargoHash = "sha256-qPKAozFXv94wgY99ugjsSuaN92SXZGgZwI2+7UlerHQ=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
Any reason why we couldn't just use `"${src}/Cargo.lock"` here? I just followed what other packages do and the Rust manual also said to copy it into the source tree.

Reply:
The `Cargo.lock` in v1.2.1 contains git dependencies, so I had to specify `cargoLock.outputHashes`. That conflicts with `cargoHash` and also requires an explicit `cargoLock.lockFile`.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#buildrustpackage-compiling-rust-applications-with-cargo-compiling-rust-applications-with-cargo